src/GHC/Util/Refact/Fixity.hs

```diff
@@ -25,15 +27,25 @@ import Data.Tuple
 
 -- | Rearrange infix expressions to account for fixity.
 -- The set of fixities is wired in and includes all fixities in base.
-applyFixities :: Anns -> Module -> (Anns, Module)
-applyFixities as m = let (as', m') = swap $ runState (everywhereM (mkM expFix) m) as
-                     in (as', m') --error (showAnnData as 0 m ++ showAnnData as' 0 m')
+applyFixities :: Anns -> [(String, Fixity)] -> Module -> (Anns, Module)
+applyFixities as fixities m = let (as', m') = swap $ runState (everywhereM (mkM (expFix fixities)) m) as
```

Comment:
The use of swap, then pattern matching immediately after, feels a bit gratuitous.